tensornetwork/matrixproductstates/base_mps.py

```diff
@@ -73,21 +75,58 @@ def __init__(self,
       raise ValueError(
           'center_position = {} not between 0 <= center_position < {}'.format(
               center_position, len(tensors)))
+    if backend is None:
+      backend = get_default_backend()
 
-    # we're no longer connecting MPS nodes because it's barely needed
+    self.backend = backend_factory.get_backend(backend)
     # the dtype is deduced from the tensor object.
-    self.nodes = [
-        Node(tensors[n], backend=backend, name='node{}'.format(n))
-        for n in range(len(tensors))
-    ]
-
-    self.connector_matrix = Node(
-        connector_matrix,
-        backend=backend) if connector_matrix is not None else connector_matrix
+    self.tensors = [self.backend.convert_to_tensor(t) for t in tensors]
+    self.connector_matrix = connector_matrix if connector_matrix is not None else connector_matrix
```

Comment:
in either case you set connector matrix equal to connector matrix.

Change this to `self.connector_matrix = connector_matrix`